index.js

```diff
@@ -269,7 +269,7 @@ ExtractTextPlugin.prototype.apply = function(compiler) {
 			async.forEach(chunks, function(chunk, callback) {
 				var extractedChunk = extractedChunks[chunks.indexOf(chunk)];
 				var shouldExtract = !!(options.allChunks || isInitialOrHasNoParents(chunk));
-				async.forEach(chunk.modules.slice(), function(module, callback) {
+				async.forEach(chunk.mapModules(function(c) { return c; }), function(module, callback) {
```

Comment:
Having just refactored this entire plugin, there are 4/5 instances of `chunk.modules.something`